src/extension.ts

```diff
@@ -151,6 +140,8 @@ export async function activate(context: vscode.ExtensionContext) {
         })
 
         toastNewUser(context, getLogger())
+
+        await loginWithMostRecentCredentials(awsContext, toolkitSettings)
```

Comment:
Is there any reason why this was bumped so far down the activation file? If we emit any metrics with an AWS account ID prior to this login, they won't have the account available.

Reply:
Yup, this lets all of the components set up (with no credentials), then react to a credentials change. This helps ensure components are handling credentials changes properly too.

Reply:
Does it affect any existing functionality though? Ultimately, (at least I'm pretty sure) this worked with the previous sequencing. Do we want to change it for the sake of changing it, especially if it could potentially affect existing metrics? Full disclosure, I haven't looked into whether or not anything is affected.

Reply:
Existing functionality isn't negatively impacted. Leaving it where it was meant adding more code to bake in knowledge of whether or not the toolkit was connected.